content/webapp/views/pages/works/work/IIIFViewer/MainViewer.tsx

```diff
@@ -462,7 +462,7 @@ const MainViewer: FunctionComponent = () => {
         mainAreaWidth,
       });
     }
-  }, [canvas]);
+  }, [canvas, shouldScrollToCanvas, currentCanvas, mainAreaWidth]);
```

Comment:
why do we need to add all these?

Reply:
I was wondering the same thing. I don't think they're necessary. Same for the `useMemo`. I'll check that everything still works well without these potentially unnecessary changes